### PR TITLE
Includes CI environment in telemetry record

### PR DESCRIPTION
**Pull request by meghfossa**
# Overview

This PR, 
- adds CI environment identifier in the telemetry record.

## Acceptance criteria

- When CLI is ran in github actions, telemetry include github_action ci indicator
- When CLI is ran in bamboo (bitbucket), telemetry include bamboo ci indicator
- When CLI is ran in azure pipeline, telemetry include azure ci indicator
- When CLI is ran in generic CI, telemetry includes ci indicator

## Testing plan

```bash
mkdir sandbox

# should be unknown_ci
CI='some-ci' ./fossa analyze sandbox --debug && cat fossa.telemetry.json | jq '.cliCIEnvironment' 

# should be github_action
GITHUB_ACTIONS=true ./fossa analyze sandbox --debug && cat fossa.telemetry.json | jq '.cliCIEnvironment'

# should be azure_pipeline
TF_BUILD=true ./fossa analyze sandbox --debug && cat fossa.telemetry.json | jq '.cliCIEnvironment'

# should be bamboo
bamboo.buildKey=123 ./fossa analyze sandbox --debug && cat fossa.telemetry.json | jq '.cliCIEnvironment'
```

Sibling PR:
- https://github.com/fossas/FOSSA/pull/8362

## Risks

N/A

## Checklist

- [x] I added tests for this PR's change (or explained in the PR description why tests don't make sense).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [x] If this change is externally visible, I updated `Changelog.md`. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [ ] If I made changes to `.fossa.yml` or `fossa-deps.{json.yml}`, I updated `docs/references/files/*.schema.json`. You may also need to update these if you have added/removed new dependency type (e.g. `pip`) or analysis target type (e.g. `poetry`).

### CODE REVIEW

File: src/Control/Carrier/Telemetry/Utils.hs [hunk 1]
```diff
@@ -91,3 +99,58 @@ getSystemInfo =
     Info.arch
     <$> Conc.getNumCapabilities
     <*> Conc.getNumProcessors
+
+lookupCIEnvironment :: IO (Maybe CIEnvironment)
```
Comment (meghfossa) [inline src/Control/Carrier/Telemetry/Utils.hs:103]: `IO` is a ramification of the telemetry effect being the top, following fmt of `getSystemInfo`...

File: src/Control/Carrier/Telemetry/Types.hs [hunk 2]
```diff
@@ -125,3 +127,21 @@ data TelemetryTimeSpent = TelemetryTimeSpent
...
+
+data CIEnvironment
+  = GithubAction
+  | AzurePipeline
+  | Bamboo
+  | UnknownCI
+  deriving (Eq, Ord, Generic)
+
+-- Modifying values below, also requires modifying value in
+-- FOSSA's server's cli telemetry type for compatibility.
```
Comment (spatten) [inline src/Control/Carrier/Telemetry/Types.hs:139]: Where are these mentioned in the server? I don't see them mentioned in https://github.com/fossas/FOSSA/pull/8362/files or elsewhere in the core codebase

**Review by spatten — APPROVED** [review]
This all looks good to me. 

I was unable to test with the `bamboo.buildKey` env variable (`.` is not allowed in env variable names in MacOS and I think Linux), but other than that it all worked.

Also, I'm assuming that it's hard to add tests for this. If not, it would be nice to have something.

File: src/Control/Carrier/Telemetry/Types.hs (hunk 2 above)
Comment (meghfossa) [reply src/Control/Carrier/Telemetry/Types.hs:139]: Ah! good catch - I had a fixed implementation approach, so this comment is not needed anymore. I will remove the comment.

**Pull request merged**